src/components/Footer.test.tsx

```diff
@@ -0,0 +1,120 @@
+import { render, screen, waitFor } from "@testing-library/react";
+import dlsLogo from "../public/dls.svg";
+import { Footer, FooterLink, FooterLinks } from "./Footer";
+
+describe("Footer", () => {
+  test("Should render blank", async () => {
+    render(<Footer logo={null} />);
+
+    await waitFor(() => {
+      expect(screen.getByTestId("footer-container")).not.toBeNull();
+      expect(screen.getByTestId("footer-link-container")).not.toBeNull();
+      // footer container only loads empty div
+      expect(
+        screen.getByTestId("footer-logo-container").childElementCount
+      ).toBe(1);
+    });
+  });
+
+  test("Should render logo only", async () => {
+    render(<Footer logo={dlsLogo} />);
+
+    await waitFor(() => {
+      expect(screen.getByRole("img")).not.toBeNull();
+    });
+  });
+
+  test("Should render copyright only", async () => {
+    render(<Footer logo={null} copyright="test copyright text" />);
+
+    await waitFor(() => {
+      expect(screen.getByText("test copyright text")).not.toBeNull();
+    });
+  });
+
+  test("Should render logo and copyright", async () => {
+    render(<Footer logo={dlsLogo} copyright="test copyright text" />);
+
+    await waitFor(() => {
+      expect(screen.getByRole("img")).not.toBeNull();
+      expect(screen.getByText("test copyright text")).not.toBeNull();
+    });
+  });
+
+  test("Should render with one link", async () => {
+    render(
+      <Footer>
+        <FooterLinks>
+          <FooterLink href="link-one-href">Link one</FooterLink>
```

Comment:
It's useful here to use a variable for "Link one", e.g. `const linkName = "Link one"` - you can then reference the variable in the test below `.toStrictEqual(linkName);`